### PR TITLE
Add gitignore file in Silicon and Platform folder

### PR DESCRIPTION
**Pull request by mauricema**
Current SBL build will show lots of FSP files as untracked files.
This patch re-orgnized the rules for .gitignore file so that it
has better control at different folder level. With these new rules
no untracked files will be shown for the FSP and VBT files.

Signed-off-by: Maurice Ma <maurice.ma@intel.com>

### CODE REVIEW

File: Silicon/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Fsp.bsf
+FspDbg.bin
+FspRel.bin
+Fsp*.h
+Fsp*.h
```
Comment (myguitar) [inline Silicon/.gitignore:5]: Please remove duplicated one.

**Review by myguitar — APPROVED** [review]
A minor comment on a duplicated one. Other than that, it looks good to me.

File: Silicon/.gitignore (hunk 1 above)
Comment (mauricema) [reply Silicon/.gitignore:5]: Sure, will do. thx!

Comment (mauricema) [reply Silicon/.gitignore]: Fixed it.

**Pull request merged**